### ISSUE TITLE
[Moon] Trying to merge desk without permission ends in infinite loop

### ISSUE THREAD

**Issue by mikolajpp** [urbit/arvo#684]
What:
If one tries to merge without permission between two moons, the source moon will enter infinite loop, even when the merge is canceled, even if the permission is turned on. 
There is no way to prevent %ames from spinning and utilizing 100%, except CTRL-C, but it will relapse back to this state after few seconds. 

Reproduction:
1. Have two moons
2. Sync a desk from one moon to the other without setting public first.
3. `What` ensues

**Comment by mikolajpp** [comment]
I haven't tested it between random ships, but I guess it will end up with the same situation.

**Comment by mikolajpp** [comment]
Short ctrl-c
```
[%swim-call-vane %a %want ~]
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/zuse
  <[2.700 7].[2.701 46]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/zuse
  <[2.701 7].[2.701 46]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/zuse
  <[2.701 12].[2.701 45]>
```

Repeated:
```
  <[1.086 7].[1.093 65]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[1.088 7].[1.093 65]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[1.088 17].[1.088 66]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[1.507 7].[1.511 32]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[1.508 7].[1.511 32]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[475 7].[485 15]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[478 13].[485 9]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[478 17].[478 40]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[478 21].[478 40]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[1.510 7].[1.511 32]>
  /~nolpem-sanrul-ponmep-litsem/home/~2018.3.24..20.31.43..e17f/sys/vane/ford
  <[1.510 23].[1.510 55]>
```

**Comment by mikolajpp** [comment]
I have hit the exact same bug, this time with moon having public desk enabled.

**Comment by mikolajpp** [comment]
Ames loop can stopped if one shuts down merging moon. 
Once it goes online again, the source moon starts spinning again.

**Comment by mikolajpp** [comment]
The merge which hanged with proper permissions, was %fine one, not %init. 
Probably worth checking out, it might turn out that for moons that one is broken regardless of permission status.

**Comment by mikolajpp** [comment]
Possible suspect has been identified, issue https://github.com/urbit/urbit/issues/936

**Comment by joemfb** [comment]
@mikolajpp, these symptoms also sound like #740 (and, I suspect, #594).

**Comment by jtobin** [comment]
Closing as it's very likely this one has been resolved by #756.
